modules/pricer/src/main/java/com/opengamma/strata/pricer/model/SabrParameters.java

```diff
@@ -0,0 +1,516 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.model;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableConstructor;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.light.LightMetaBean;
+
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.value.ValueDerivatives;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.market.ValueType;
+import com.opengamma.strata.market.curve.ConstantCurve;
+import com.opengamma.strata.market.curve.Curve;
+import com.opengamma.strata.market.curve.CurveInfoType;
+import com.opengamma.strata.market.curve.Curves;
+import com.opengamma.strata.market.curve.InterpolatedNodalCurve;
+import com.opengamma.strata.market.param.ParameterMetadata;
+import com.opengamma.strata.market.param.ParameterPerturbation;
+import com.opengamma.strata.market.param.ParameterizedData;
+import com.opengamma.strata.market.param.ParameterizedDataCombiner;
+
+/**
+ * The volatility surface description under SABR model.
+ * <p>
+ * This is used in interest rate modeling.
+ * Each SABR parameter is a {@link Curve} defined by expiry.
+ * <p>
+ * The implementation allows for shifted SABR model.
+ * The shift parameter is also {@link Curve} defined by expiry.
+ */
+@BeanDefinition(style = "light")
+public final class SabrParameters
```

Comment:
We should probably review the name of this class and 'SabrInterestRateParameters'. Maybe names like 'SabrShiftedSurfaceParameters' and 'SabrShiftedCurveParameters' would be more coherent and explicit.